Visibility Checkbox Updated

Mohamed-A-Mohamed opened the pull request:
Added the ability to hide the list of check boxes in the instructor view - Upadated on 10/19/2017:
![hide-show](https://user-images.githubusercontent.com/19738915/31803427-a6ebf0ee-b521-11e7-9a75-7d004342143a.gif)
---------------------

**# Updated so now everything doesn't get shifted:**
![hide-show2](https://user-images.githubusercontent.com/19738915/31865920-1a123d38-b745-11e7-823c-9d13b9b10d4d.gif)

mohamedamohamed committed:
Added CSS for check box so that it is in-line with the title upon creation. Also added the ability to hide and show the list of check boxes in the professor view.

mohamedamohamed committed:
Merged quizzard and added the ability to toggle the list of checkboxes in the instructor view

Mohammed-JS-Khan commented on public/css/admin.css:
```diff
@@ -29,6 +29,12 @@ body {
+.check-box {
+  text-align: left;
+  display: block;
+
```
extra line

Mohamed-A-Mohamed commented:
This pull request takes care of enhancement #42

Mohammed-JS-Khan commented on views/question-table.pug:
```diff
@@ -1,7 +1,8 @@
     tr
-      th Visible
+      th
+        button(id='vis' type='button' value='true' onClick='toggleButtonVisibility()') Visiblity
```
Maybe we can label the column (or the button) something like toggle visibility.

Mohammed-JS-Khan replied:
Or maybe we can leave the column header there and append a really small hyperlinked text that says (show/hide) that will toggle the checkboxes themselves. DIVA requirement though. Good work otherwise.

Mohamed-A-Mohamed replied:
I see what you're saying, it would be a nice touch.

Mohammed-JS-Khan commented:
Looks good to me, just some minor things to fix and mabye a team discussion on the visibility portion. Good work though, I really like the screenshots you provided! :)

esteeroramy commented on public/js/admin.js:
```diff
@@ -659,6 +659,17 @@ var sortAccountsTable = function(type) {
     });
 }
 
+//toggles the view of Visiblility button in the Question-Table View
```
space after // please

Mohamed-A-Mohamed replied:
Fixed

esteeroramy replied:
not fixed

Mohamed-A-Mohamed replied:
It's now fixed for sure :+1:  :smile:

esteeroramy commented on public/js/admin.js:
```diff
@@ -659,6 +659,17 @@ var sortAccountsTable = function(type) {
 
+//toggles the view of Visiblility button in the Question-Table View
+var toggleButtonVisibility = function(){
+    if (document.getElementById('vis').value === 'true') {
```
seems odd that true and false are between quotes

Mohammed-JS-Khan replied:
@esteeroramy it doesn't really matter, it was just used as a flag since we had to put a string value there.

Mohamed-A-Mohamed replied:
Removed the quotes

esteeroramy commented on views/question-table.pug:
```diff
@@ -1,7 +1,8 @@
     tr
-      th Visible
+      th
+        button(id='vis' type='button' value='true' onClick='toggleButtonVisibility()') Visiblity
```
can we use a button from the material design library

Mohamed-A-Mohamed replied:
I changed it to a switch from the material design library

esteeroramy commented on views/question-table.pug:
```diff
@@ -26,3 +27,4 @@ table.table
             span.glyphicon.glyphicon-pencil(aria-hidden='true')
           button.btn.btn-default.delete-button(id='delete-' + q.id type='button' aria-label='Delete')
             span.glyphicon.glyphicon-remove(aria-hidden='true')
+
```
you can remove this extra line

Mohamed-A-Mohamed replied:
Fixed

ahmedqarmout2 commented:
Please update the picture after you use the new design. Also plz use the switch

Momo - I changed it to a switch :+1:

ahmedqarmout2 reviewed (approved):
Address all the comments, and update the design

mohamedamohamed committed:
Merged changes for new CSS and made button to hide/show list of checkboxes to a switch

mohamedamohamed committed:
if getVisibleValue() == 'true'
Merge branch 'develop' of https://github.com/quizzard-edu/quizzard into develop

Merging the updates from PR #51 "New Question Structure"

esteeroramy commented on public/css/admin.css:
```diff
@@ -0,0 +1,89 @@
+body {
```
I don't think that this file is a part of your change

Mohamed-A-Mohamed replied:
Ya, I'm not sure how this happened. I was just removing an extra line at the end of a file.

Mohamed-A-Mohamed replied:
The reason this happened is because Ahmed removed all the css because it's all in the materialize library. I didn't realize this when merging his changes. This has now been fixed.

esteeroramy commented:
I think there is some sort of padding around the check mark that is making the items in the list expand, please take a look at that

Momo - I will look into this.

mohamedamohamed committed:
Added css so that when you hide/show the list of checkboxes it doesn't shift the question table view

mohamedamohamed committed:
if getVisibleValue() == 'true'
Merge branch 'develop' of https://github.com/quizzard-edu/quizzard into develop

Merging as new features such as toasts and professor quesiton rating have been added.

esteeroramy commented on public/js/admin.js:
```diff
@@ -686,6 +686,15 @@ var sortAccountsTable = function(type) {
 
+// Toggles the view of the Visibility Checkboxes in the Question-Table View
+var toggleButtonVisibility = function(){
+    if (document.getElementById('sw').checked === true) {
```
`if (document.getElementById('sw').checked) {`

Mohamed-A-Mohamed replied:
fixed

esteeroramy commented on public/js/admin.js:
```diff
@@ -686,6 +686,15 @@ var sortAccountsTable = function(type) {
+var toggleButtonVisibility = function(){
+    if (document.getElementById('sw').checked === true) {
+        $('.visbox').show();
+    } else{
```
space after `else`

Mohamed-A-Mohamed replied:
fixed

mohamedamohamed committed:
Changed if statement in toggleButtonVisibility function so it doesn't check equal to True, added space after else

Mohammed-JS-Khan reviewed (approved):
looks good to me
